pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -96,25 +98,19 @@ type outputEventFn func(e *roachpb.RangeFeedEvent) error
 
 // CatchUpScan iterates over all changes in the configured key/time span, and
 // emits them as RangeFeedEvents via outputFn in chronological order.
+//
+// MVCC range tombstones are emitted at their start key (in chronological
+// order), and may thus be output before subsequent keys that they cover. For
+// example, the range key [a-f)@3 will be emitted before b@2, even though the
```

Comment:
It's strange seeing this since above we say events emitted in chronological order...

Reply:
Keeping strict chronological ordering below range tombstones is going to kill performance. Consider the following example:

```
5   a5  b5  c5
4   o-----------o
3   a3  b3  c3
2   o-----------o
1   a1  b1  c1
    a   b   c   d
```

If we were to keep strict ordering, then we would have to emit `a1,b1,c1` first, then `[a-d)@2`, then `a3,b3,c3`, and so on. Going back and forth to scan these slices is essentially going to be random access, which is extremely inefficient. Instead, range keys are emitted before the point keys they cover, which is the optimally efficient way to process them -- e.g. `[a-d)@2,[a-d)@4,a1,a3,a5,b1,b3,b5,...`.